server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -221,6 +232,35 @@ protected void masterOperation(
 
             listener.delegateFailureAndWrap((delegate, statsResponse) -> {
 
+                AutoShardingResult rolloverAutoSharding = null;
+                final IndexAbstraction indexAbstraction = clusterState.metadata()
+                    .getIndicesLookup()
+                    .get(rolloverRequest.getRolloverTarget());
+                if (indexAbstraction.getType().equals(IndexAbstraction.Type.DATA_STREAM)) {
+                    DataStream dataStream = (DataStream) indexAbstraction;
+                    final Optional<IndexStats> indexStats = Optional.ofNullable(statsResponse)
+                        .map(stats -> stats.getIndex(dataStream.getWriteIndex().getName()));
+
+                    Double writeLoad = indexStats.map(stats -> {
+                        IndexingStats indexing = stats.getTotal().getIndexing();
+                        if (indexing != null) {
+                            return indexing.getTotal().getWriteLoad();
+                        } else {
+                            return null;
+                        }
+                    }).orElse(null);
+
+                    rolloverAutoSharding = dataStreamAutoShardingService.calculate(clusterState, dataStream, writeLoad);
+                    logger.debug("data stream auto sharding result is [{}]", rolloverAutoSharding);
```

Comment:
Should this log statement include the data stream in question? Otherwise it's just the result without knowing which data stream we're talking about, right?